src/main/java/pl/betse/beontime/timesheet/service/TimeEntryService.java

```diff
@@ -0,0 +1,233 @@
+package pl.betse.beontime.timesheet.service;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.web.bind.annotation.RequestMethod;
+import pl.betse.beontime.timesheet.bo.StatusBo;
+import pl.betse.beontime.timesheet.bo.TimeEntryBo;
+import pl.betse.beontime.timesheet.entity.StatusEntity;
+import pl.betse.beontime.timesheet.entity.TimeEntryEntity;
+import pl.betse.beontime.timesheet.exception.*;
+import pl.betse.beontime.timesheet.mapper.StatusMapper;
+import pl.betse.beontime.timesheet.mapper.TimeEntryMapper;
+import pl.betse.beontime.timesheet.repository.StatusRepository;
+import pl.betse.beontime.timesheet.repository.TimeEntryRepository;
+
+import java.time.LocalDate;
+import java.time.temporal.IsoFields;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Service
+public class TimeEntryService {
+
+    private final TimeEntryRepository timeEntryRepository;
+    private final TimeEntryMapper timeEntryMapper;
+    private final StatusRepository statusRepository;
+    private final StatusMapper statusMapper;
+    private final String SAVED_STATUS = "SAVED";
+    private final String REJECTED_STATUS = "REJECTED";
+
+
+    public TimeEntryService(TimeEntryRepository timeEntryRepository, TimeEntryMapper timeEntryMapper, StatusRepository statusRepository, StatusMapper statusMapper) {
+        this.timeEntryRepository = timeEntryRepository;
+        this.timeEntryMapper = timeEntryMapper;
+        this.statusRepository = statusRepository;
+        this.statusMapper = statusMapper;
+    }
+
+    public List<List<TimeEntryBo>> findByUserGuidAndWeek(String userGuid, String week) {
+        List<TimeEntryBo> timeEntryBoList = timeEntryRepository.findByUserGuidAndWeek(userGuid, week).stream()
+                .map(timeEntryMapper::fromEntityToBo)
+                .collect(Collectors.toList());
+        List<String> projectNames = new ArrayList<>();
+        resolveProjectGuidFromBoList(timeEntryBoList, projectNames);
+        List<List<TimeEntryBo>> allProjectsForWeekList = new ArrayList<>();
+        projectNames.forEach(projectGuid ->
+                allProjectsForWeekList.add(timeEntryRepository.findByUserGuidAndProjectGuidAndWeek(userGuid, projectGuid, week).stream()
+                        .map(timeEntryMapper::fromEntityToBo)
+                        .collect(Collectors.toList()))
+        );
```

Comment:
this logic is totally unclear to me. Please try to simplify it. You are doing a lot of DB requests and I am not sure if they are necessary.